test/conftest.py

```diff
@@ -0,0 +1,159 @@
+    async_mock = AsyncMock()
+    mocker.patch('api.db.Database.find_one',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def mock_get_current_user(mocker):
+    """
+    Mocks async call to Authentication class method
+    used to get current user
+    """
+    async_mock = AsyncMock()
+    user = User(username='bob',
+                hashed_password='$2b$12$CpJZx5ooxM11bCFXT76/z.o6HWs2sPJy4iP8.'
+                                'xCZGmM8jWXUXJZ4K',
+                active=True)
+    mocker.patch('api.auth.Authentication.get_current_user',
+                 side_effect=async_mock)
+    async_mock.return_value = user
+    return async_mock
+
+
+@pytest.fixture
+def mock_init_sub_id(mocker):
+    """Mocks async call to PubSub method to initialize subscription id"""
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub._init_sub_id',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def mock_listen(mocker):
+    """Mocks async call to listen method of PubSub"""
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub.listen',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def mock_publish_cloudevent(mocker):
+    """
+    Mocks async call to PubSub class method
+    used to publish cloud event
+    """
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub.publish_cloudevent',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def mock_pubsub(mocker):
+    """Mocks `_redis` member of PubSub class instance"""
+    pubsub = PubSub()
+    redis_mock = fakeredis.aioredis.FakeRedis()
+    mocker.patch.object(pubsub, '_redis', redis_mock)
+    return pubsub
+
+
+@pytest.fixture
+def mock_pubsub_subscriptions(mocker):
+    """Mocks `_redis` and `_subscriptions` member of PubSub class instance"""
+    pubsub = PubSub()
+    redis_mock = fakeredis.aioredis.FakeRedis()
+    mocker.patch.object(pubsub, '_redis', redis_mock)
+    subscriptions_mock = dict({1: pubsub._redis.pubsub()})
+    mocker.patch.object(pubsub, '_subscriptions', subscriptions_mock)
+    return pubsub
+
+
+@pytest.fixture
+def mock_subscribe(mocker):
+    """Mocks async call to subscribe method of PubSub"""
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub.subscribe',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def mock_unsubscribe(mocker):
+    """Mocks async call to unsubscribe method of PubSub"""
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub.unsubscribe',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture
+def get_bearer_token():
+    """Returns bearer token to be used for Authentication"""
+    return "Bearer \
+            eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9. \
+            eyJzdWIiOiJib2IifQ.ci1smeJeuX779PptTkuaG1S \
+            Edkp5M1S1AgYvX8VdB20"
```

Comment:
Why is this a fixture and not a regular function?

Reply:
If I define it as a regular function, I will need to import it by `from test.conftest import get_bearer_token`.
In the case of fixture, it automatically gets imported by defining in conftest.py just like other fixtures.

Reply:
OK, but I don't see a good reason for not just importing the module and using it like any regular Python code.

Reply:
In fact, the string could also be defined directly in the module e.g. `conftest.BEARER_TOKEN`.